src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/IncentivesService.kt

```diff
@@ -35,19 +35,35 @@ class IncentivesService(
         offenderBooking = offenderBooking,
         sequence = incentiveSequence
       )
-    )?.let { mapIncentiveModel(it) }
+    )?.let {
+      // determine if this incentive is the current IEP for the booking
+      val currentIep =
+        incentiveRepository.findFirstById_offenderBookingOrderByIepDateDescId_SequenceDesc(offenderBooking)
+          ?.let { current -> current == it } ?: false
```

Comment:
i *think possibly this is equivalent: 
val currentIep = it ==
        incentiveRepository.findFirstById_offenderBookingOrderByIepDateDescId_SequenceDesc(offenderBooking)

Reply:
yes - thanks, makes it simpler